File: src/main/kotlin/com/stannis/dataModel/statementTypes/SimpleDeclSpecifier.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.DeclarationSpecifierParent
import com.stannis.dataModel.Statement

data class SimpleDeclSpecifier(
    override val `$type`: String? = "SimpleDeclSpecifier",
    var type: Int?,
    var isSigned: Boolean,
    var isUnsigned: Boolean,
    var isShort: Boolean,
    var isLong: Boolean,
    var isLongLong: Boolean,
    var isComplex: Boolean,
    var isImaginary: Boolean,
    var fDeclTypeExpression: Statement?,
    var isConstant: Boolean,
    var isExplicit: Boolean,
    var isFriend: Boolean,
    var isInline: Boolean,
    var isRestrict: Boolean,
    var isThreadLocal: Boolean,
    var isVirtual: Boolean,
    var isVolatile: Boolean,
    var declarationSpecifier: String?
) : Statement, DeclarationSpecifierParent

File: Interpreter/Interpreter/Models/serialize/statementTypes/SimpleDeclSpecifier.cs
namespace Interpreter.Models.serialize.statementTypes{
    public class SimpleDeclSpecifier : IStatement
    {
        public string declarationSpecifier { set; get; }
        public int type { set; get; }
        public bool isSigned { set; get; }
        public bool isUnsigned { set; get; }
        public bool isShort { set; get; }
        public bool isLong { set; get; }
        public bool isLongLong { set; get; }
        public bool isComplex { set; get; }
        public bool isImaginary { set; get; }
        public IStatement fDeclTypeExpression { set; get; }
        public bool isConstant { set; get; }
        public bool isExplicit { set; get; }
        public bool isFriend { set; get; } 
        public bool isInline { set; get; }
        public bool isRestrict { set; get; }
        public bool isThreadLocal { set; get; }
        public bool isVirtual { set; get; }
        public bool isVolatile { set; get; }
    }
}